check-commit-email.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/python3
+
+import os
+import subprocess
+from email.utils import parseaddr
+
+is_github_actions = os.getenv("GITHUB_ACTIONS", "") == "true"
+
+
+def get_merge_commit_range():
+    # The head revision is a synthesised merge commit, merging the
+    # proposed branch into the destination branch.  So the first
+    # parent is our destination, and the second is our proposal.
+    lines = subprocess.check_output(
+        ["git", "cat-file", "-p", "@"], text=True
+    ).splitlines()
+    parents = [
+        line[len("parent ") :].strip() for line in lines if line.startswith("parent ")
+    ]
+    if len(parents) != 2:
+        raise RuntimeError("expected two parents, but got {}".format(parents))
+    dest, proposed = parents
+
+    return "{}..{}".format(dest, proposed)
+
+
+if __name__ == "__main__":
+    commitrange = "HEAD"
+    if is_github_actions:
+        # in github actions the HEAD commit is a merge commit and it doesn't
+        # work to just specify HEAD, so instead use the merge commit range
+        commitrange = get_merge_commit_range()
+
+    for line in subprocess.check_output(
+        ["git", "log", "--format=format:%h,%ce%n%h,%ae", commitrange],
+        text=True,
+    ).split("\n"):
+        parsed = line.split(",", 1)
+        commithash = parsed[0]
+        potentialemail = parsed[1]
+        if potentialemail == "":
+            continue
+        name, addr = parseaddr(potentialemail)
+        if addr == "":
+            print(
```

Comment:
(nitpick^9) I prefer a different forming style - `go fmt` for the rescue. But ignore that please because it really does not matter :)

Reply:
but go fmt does not work on python files ?

also this is the suggested formatting as specified by `black`

Reply:
yes, this looks like a black's doing, which is fine as long as we continue using it

Reply:
Aha, nice. That's fine then, if it's some sort of auto-format I'm fine with that.